src/Cli/dotnet/Commands/Test/MTP/TestApplicationActionQueue.cs

```diff
@@ -61,6 +61,12 @@ private async Task Read(Func<TestApplication, Task<int>> action, BuildOptions bu
                 var testApp = new TestApplication(module, buildOptions, testOptions, output);
                 using (testApp)
                 {
+                    // TODO: If this throws, we will loose "one" degree of parallelism.
```

Comment:
is this TODO expected to be fixed in this PR? or is there a bug to track this TODO?

the MTP client for VS does bubble up the abort. so that if one test application fails to run it will bubble it up for all applications. however, we do this by setting a shared state, while the ActionBlock will try/catch any exceptions to prevent tasks from being stopped because of one client crashing